### PR TITLE
External state access

### PR DESCRIPTION
**Pull request by mbialon**
# Description of the change

Describing how to use external state access (a.k.a. remote terraform backend).

- Add documentation about external state access
- Remove the reasons not to let Spacelift manage state, invalidated by external state access
- Mention external state access in exporting stack state section

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [x] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [x] The preview looks fine.
- [x] The tests pass.
- [x] The commit history is clean and meaningful.
- [x] The pull request is opened against the `main` branch.
- [x] The pull request is no longer marked as a draft.
- [x] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift (not required for Spacelift employees).

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

_Spacelift employees should request reviews from the relevant engineers and are allowed to merge pull requests after they got at least one approval._

Thank you for your contribution! 🙇

### CODE REVIEW

File: docs/vendors/terraform/version-management.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ If you're creating stacks programmatically but intend to make independent change
...
 
 !!! warning
-    The newest Terraform version supported by Spacelift may lag a bit behind the latest available Terraform release. We err on the side of caution and thus separately verify each version to ensure that it works as expected and that our code is compatible with its output and general behavior. We're trying to catch up roughly within a week but **may temporarily blacklist a faulty version**. If you need a compatibility check and a bump sooner than that, please get in touch with our support.
+The newest Terraform version supported by Spacelift may lag a bit behind the latest available Terraform release. We err on the side of caution and thus separately verify each version to ensure that it works as expected and that our code is compatible with its output and general behavior. We're trying to catch up roughly within a week but **may temporarily blacklist a faulty version**. If you need a compatibility check and a bump sooner than that, please get in touch with our support.
```
Comment (jmfontaine) [inline docs/vendors/terraform/version-management.md]: This will break the callout.

File: docs/vendors/terraform/version-management.md [hunk 2]
```diff
@@ -42,4 +42,4 @@ Once we apply a run with a particular version of Terraform, we set it on the sta
 In order to migrate a [stack](../../concepts/stack/README.md) to a newer version of Terraform, we suggest opening a feature branch bumping the version through [runtime configuration](../../concepts/configuration/runtime-configuration/README.md#terraform_version-setting). Open a Pull Request in GitHub from the feature branch to the tracked branch to easily get a link to your proposed run and see if everything looks good. If it does, merge your Pull Request and enjoy working with the latest and greatest version of Terraform. Otherwise, try making necessary changes until your code is working or postpone the migration until you have the bandwidth to do so.
 
 !!! info
-    In general, we suggest to try and keep up with the latest Terraform releases. **The longer you wait, the more serious is the migration work going to be**. Terraform evolves, providers evolve, external APIs evolve and so should your code.
```
Comment (jmfontaine) [inline docs/vendors/terraform/version-management.md]: This will break the callout.

File: nav.yaml [hunk 3]
```diff
@@ -3,146 +3,147 @@ nav:
...
-    - Spaces:
-        - concepts/spaces/README.md
-        - concepts/spaces/access-control.md
-        - concepts/spaces/creating-a-space.md
-        - concepts/spaces/moving-a-space-or-an-entity.md
-        - concepts/spaces/deleting-a-space.md
-        - concepts/spaces/structuring-your-spaces-tree.md
-        - concepts/spaces/migrating-out-of-the-legacy-space.md
-    - concepts/vcs-agent-pools.md
+      - Stack:
```
Comment (jmfontaine) [inline nav.yaml:6]: Why change the indentation?

File: docs/vendors/terraform/external-state-access.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+}
+```
+
+1. The hostname of the Spacelift system.
+2. The name of the account.
+3. The ID of the stack you wish to retrieve outputs from.
+
+## Offline state access
+
+Given an account called _spacecorp_, and a stack named _deep-thought_, having the following Terraform configuration.
```
Comment (jmfontaine) [inline docs/vendors/terraform/external-state-access.md:67]: Would not this syntax look better here (and for other values on that page)?

```suggestion
Given an account called `spacecorp`, and a stack named `deep-thought`, having the following Terraform configuration.
```

File: docs/vendors/terraform/external-state-access.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+```shell
+terraform login spacelift.io
+```
+
+Next, you need a [remote backend](https://developer.hashicorp.com/terraform/language/settings/backends/remote){: rel="nofollow"} configuration.
+
+```terraform
+terraform {
+  backend "remote" {
+    hostname     = "spacelift.io" # 1
```
Comment (jmfontaine) [inline docs/vendors/terraform/external-state-access.md:86]: Did you try using Code Annotations?

https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-annotations

File: nav.yaml (hunk 3 above)
Comment (mbialon) [reply nav.yaml]: VS Code decided to reformat the whole file. Fixed.

File: docs/vendors/terraform/external-state-access.md (hunk 5 above)
Comment (mbialon) [reply docs/vendors/terraform/external-state-access.md]: It's much better. Thanks for the hint.

**Review by jmfontaine — DISMISSED** [review]

**Comment by mbialon** [comment]
@jmfontaine I've squashed the commits.

**Review by jmfontaine — APPROVED** [review]

**Pull request merged**